classes/boostnavbar.php

```diff
@@ -0,0 +1,233 @@
+                case 'group-overview':
+                case 'group-assign':
+                    // Remove the 'Groups' navbar node in the Groupings, Grouping, group Overview and Assign pages.
+                    $this->remove('groups');
+                case 'backup-backup':
+                case 'backup-restorefile':
+                case 'backup-copy':
+                case 'course-reset':
+                    // Remove the 'Import' navbar node in the Backup, Restore, Copy course and Reset pages.
+                    $this->remove('import');
+                case 'course-user':
+                    $this->remove('mygrades');
+                    $this->remove('grades');
+            }
+        }
+
+        // Remove 'My courses' if we are in the module context.
+        if ($this->page->context->contextlevel == CONTEXT_MODULE) {
+            $this->remove('mycourses');
+            $this->remove('courses');
+            // Remove the course category breadcrumb node.
+            $this->remove($this->page->course->category, \breadcrumb_navigation_node::TYPE_CATEGORY);
+            $courseformat = course_get_format($this->page->course)->get_course();
+            // Section items can be only removed if a course layout (coursedisplay) is not explicitly set in the
+            // given course format or the set course layout is not 'One section per page'.
+            $removesections = !isset($courseformat->coursedisplay) ||
+                $courseformat->coursedisplay != COURSE_DISPLAY_MULTIPAGE;
+            if ($removesections) {
+                // If the course sections are removed, we need to add the anchor of current section to the Course.
+                $coursenode = $this->get_item($this->page->course->id);
+                if (!is_null($coursenode) && $this->page->cm->sectionnum !== null) {
+                    $coursenode->action = course_get_format($this->page->course)->get_view_url($this->page->cm->sectionnum);
+                }
+            }
+        }
+
+        if ($this->page->context->contextlevel == CONTEXT_SYSTEM) {
+            // Remove the navbar nodes that already exist in the secondary navigation menu.
+            $this->remove_items_that_exist_in_navigation($PAGE->secondarynav);
+        }
+
+        // Set the designated one path for courses.
+        $mycoursesnode = $this->get_item('mycourses');
+        if (!is_null($mycoursesnode)) {
+            $url = new \moodle_url('/my/courses.php');
+            $mycoursesnode->action = $url;
+            $mycoursesnode->text = get_string('mycourses');
+        }
+
+        $this->remove_no_link_items($removesections);
+
+        // Don't display the navbar if there is only one item. Apparently this is bad UX design.
+        if ($this->item_count() <= 1) {
+            $this->clear_items();
+            return;
+        }
+
+        // Make sure that the last item is not a link. Not sure if this is always a good idea.
+        if (false) {
```

Comment:
Yes, right.
I just wanted to prevent the last item link function being removed.
Probably with the new "Change nothing" I have to leave this in and make sure it is still done when no divergent behaviour in regards to Boost core is desired.
I hesitated to make "Remove breadcrumb navigation lase item link" "yes/no" a configurable option by the way but then decided against it. This can still be done later on.